internal/campaigns/archive_fetcher.go

```diff
@@ -34,7 +34,14 @@ func (wc *WorkspaceCreator) Create(ctx context.Context, repo *graphql.Repository
 	}
 
 	if !exists {
-		if err := fetchRepositoryArchive(ctx, wc.client, repo, path); err != nil {
+		err = fetchRepositoryArchive(ctx, wc.client, repo, path)
+		if err != nil {
+			if errors.Is(err, context.Canceled) || errors.Cause(err) == context.Canceled {
```

Comment:
Why not always remove in the case of an error?

Reply:
You're right. There might be other errors that leave a partial file lying around.